File: models/creator-card.js
const { ModelSchema, SchemaTypes, DatabaseModel } = require('@app-core/mongoose');

const modelName = 'creatorCards';

const schemaConfig = {
  _id: { type: SchemaTypes.ULID },
  title: { type: SchemaTypes.String },
  description: { type: SchemaTypes.String },
  slug: { type: SchemaTypes.String, unique: true, index: true },
  creator_reference: { type: SchemaTypes.String, index: true },
  links: { type: SchemaTypes.Array },
  service_rates: { type: SchemaTypes.Mixed },
  status: { type: SchemaTypes.String, index: true },
  access_type: { type: SchemaTypes.String, index: true },
  access_code: { type: SchemaTypes.String },
  created: { type: SchemaTypes.Number },
  updated: { type: SchemaTypes.Number },
};

const modelSchema = new ModelSchema(schemaConfig, { collection: modelName });

module.exports = DatabaseModel.model(modelName, modelSchema, { paranoid: true });

File: specs/creator-cards/data/creator-card.go
import ../commons.go

CreatorCard {
  _id string<isUnique|indexed>
  title string
  description? string
  slug string<isUnique|indexed>
  creator_reference string<indexed>
  links[]? {
    title string
    url string
  }
  service_rates? {
    currency string
    rates[] {
      name string
      description string
      amount number
    }
  }
  status string
  access_type string
  access_code? string
  ...common
}
